src/Core/TestSuite/ContainerStubTrait.php

```diff
@@ -144,7 +145,12 @@ public function modifyContainer(EventInterface $event, ContainerInterface $conta
         }
         foreach ($this->containerServices as $key => $factory) {
             if ($container->has($key)) {
-                $container->extend($key)->setConcrete($factory);
+                try {
+                    $container->extend($key)->setConcrete($factory);
+                } catch (NotFoundException $e) {
+                    $container->add($key, $factory);
+                    $container->extend($key)->setConcrete($factory);
```

Comment:
This line seems unnecessary, as the definition with the mock has already been added. And I guess a test for that would be nice too :)

Reply:
Sorry @ndm2 I don't follow.

Reply:
`$container->add($key, $factory);` adds the mock to the container, you don't have to extend that definition, it already has the required concrete attached.